pandas/core/groupby/generic.py

```diff
@@ -1546,14 +1546,14 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             if self._grouper.nkeys > 1:
                 # test_groupby_as_index_series_scalar gets here with 'not self.as_index'
                 return self._python_agg_general(func, *args, **kwargs)
-            elif args or kwargs:
+            elif (args or kwargs) and (len(self._obj_with_exclusions.columns) == 1):
```

Comment:
Why do you need this extra condition? With this modification, I am thinking we should hold off on merging this PR. `_aggregate_frame` passes the entire DataFrame rather than column-by-column that the rest of the code paths do. So this could cause breakage in user code.

Reply:
Thanks Richard, 

That was the intent, because the bug currently occurs when multiple columns to be aggregated are passed into `_aggregate_frame`, the aggregation operates on all columns and provides duplicate (and incorrect) values such as this (full example in #58146):

```
A	B
groupby1		
diamond	13.25	13.25
spade	27.00	27.00
```

Instead of using that method when there are multiple columns, I thought passing it to the next code path that does column-by-column would address the issue.

That said, I don't have the skills and knowledge of the codebase the maintainers have to fully appreciate the extent of impact this change might have on the users' codebase. Just thought I might put this idea forward as a change that would address the issue of multiple columns getting aggregated together.

Reply:
I see - thanks! Yes, we discussed this on the last dev meeting. I think we want to look to see if we can deprecate this behavior instead of changing it as a bug fix because it has the risk of quietly changing the value in the result without failing.

Reply:
I see, no worries! Just thought it might be an appropriate workaround

I was hoping to get a workaround in so I might be able to work on the `numeric_only` support as having a kwarg pushes the code down this pathway more frequently. Which then would allow me to address an issue in Geopandas. It sounds like it might be best to wait for the deprecation before working on the downstream items.

I'l update my PR to only address the `GroupByApply`.

Thanks for your time and feedback.